pkg/watch/module_release_meta_change.go

```diff
@@ -0,0 +1,174 @@
+	client.Reader
+
+	CreateFunc func(context.Context, event.TypedCreateEvent[object], workqueue.TypedRateLimitingInterface[request])
+
+	UpdateFunc func(context.Context, event.TypedUpdateEvent[object], workqueue.TypedRateLimitingInterface[request])
+
+	DeleteFunc func(context.Context, event.TypedDeleteEvent[object], workqueue.TypedRateLimitingInterface[request])
+
+	GenericFunc func(context.Context, event.TypedGenericEvent[object], workqueue.TypedRateLimitingInterface[request])
+}
+
+func NewModuleReleaseMetaEventHandler(handlerClient ChangeHandlerClient) *ModuleReleaseMetaEventHandler {
+	return &ModuleReleaseMetaEventHandler{Reader: handlerClient}
+}
+
+// Create handles Create events.
+func (m TypedModuleReleaseMetaEventHandler[object, request]) Create(ctx context.Context, event event.CreateEvent,
+	rli workqueue.TypedRateLimitingInterface[reconcile.Request],
+) {
+	handleEvent(ctx, event, rli, m.Reader)
+}
+
+// Delete handles Delete events.
+func (m TypedModuleReleaseMetaEventHandler[object, request]) Delete(ctx context.Context, event event.DeleteEvent,
+	rli workqueue.TypedRateLimitingInterface[reconcile.Request],
+) {
+	handleEvent(ctx, event, rli, m.Reader)
+}
+
+// Generic handles generic events.
+func (m TypedModuleReleaseMetaEventHandler[object, request]) Generic(ctx context.Context, event event.GenericEvent,
+	rli workqueue.TypedRateLimitingInterface[reconcile.Request],
+) {
+	handleEvent(ctx, event, rli, m.Reader)
+}
+
+// Update handles Update events.
+func (m TypedModuleReleaseMetaEventHandler[object, request]) Update(ctx context.Context, event event.UpdateEvent,
+	rli workqueue.TypedRateLimitingInterface[reconcile.Request],
+) {
+	kymaList, err := getKymaList(ctx, m.Reader)
+	if err != nil {
+		return
+	}
+
+	oldModuleReleaseMeta, ok := event.ObjectOld.(*v1beta2.ModuleReleaseMeta)
+	if !ok {
+		return
+	}
+	newModuleReleaseMeta, ok := event.ObjectNew.(*v1beta2.ModuleReleaseMeta)
+	if !ok {
+		return
+	}
+	diff := DiffModuleReleaseMetaChannels(oldModuleReleaseMeta, newModuleReleaseMeta)
+
+	affectedKymas := GetAffectedKymas(kymaList, newModuleReleaseMeta.Spec.ModuleName, diff)
+
+	requeueKymas(rli, affectedKymas)
+}
+
+// DiffModuleReleaseMetaChannels determines the difference between the old and new ModuleReleaseMeta channels. It returns
+// a map of the channels that have been updated or added.
+func DiffModuleReleaseMetaChannels(oldModuleReleaseMeta, newModuleReleaseMeta *v1beta2.ModuleReleaseMeta) map[string]v1beta2.ChannelVersionAssignment {
+	diff := make(map[string]v1beta2.ChannelVersionAssignment)
+
+	oldChannels := make(map[string]v1beta2.ChannelVersionAssignment)
+	for _, channel := range oldModuleReleaseMeta.Spec.Channels {
+		oldChannels[channel.Channel] = channel
+	}
+
+	for _, newChannel := range newModuleReleaseMeta.Spec.Channels {
+		oldChannel, ok := oldChannels[newChannel.Channel]
+		if !ok || oldChannel.Version != newChannel.Version {
+			diff[newChannel.Channel] = newChannel
+		}
+	}
+
```

Comment:
shouldn't we also return a diff if a channel was removed to trigger a requeue?

Reply:
Will implement it!